unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Video/DCLVideoTexture.cs

```diff
@@ -228,15 +238,21 @@ private void CalculateVideoVolumeAndPlayStatus()
             UpdateVolume();
         }
 
+        private void OnVirtualAudioMixerChangedValue(float currentValue, float previousValue) {
+            UpdateVolume();
+        }
+
         private void UpdateVolume()
         {
             if (texturePlayer == null)
                 return;
 
             float targetVolume = 0f;
 
-            if (CommonScriptableObjects.rendererState.Get() && IsPlayerInSameSceneAsComponent((CommonScriptableObjects.sceneID.Get())))
-                targetVolume = baseVolume * distanceVolumeModifier * Settings.i.generalSettings.sfxVolume;
+            if (CommonScriptableObjects.rendererState.Get() && IsPlayerInSameSceneAsComponent((CommonScriptableObjects.sceneID.Get()))) {
+                targetVolume = baseVolume * distanceVolumeModifier;
+                targetVolume *= Utils.ToVolumeCurve(DataStore.i.virtualAudioMixer.sceneSFXVolume.Get() * Settings.i.currentAudioSettings.sceneSFXVolume * Settings.i.currentAudioSettings.masterVolume);
+            }
```

Comment:
Can you spread this code a bit more using temporary variables if necessary? It's getting very hard to read as it is right now.

Reply:
Split it into temporary variables.
![image](https://user-images.githubusercontent.com/1711394/128317432-1902cbb4-b6f6-4f31-8eca-0514ea09c200.png)